### ISSUE TITLE
Advanced peripherals has failed to load correctly

### ISSUE THREAD

**Issue by Kolterdyx** [Seniorendi/AdvancedPeripherals#233]
### Describe

When trying to run Minecraft with CC: Tweaked and Advanced peripherals, the game crashes while loading, with the following error:
`java.lang.reflect.InvocationTargetException: null`

This is the error from `latest.log`

```
[00:18:46] [modloading-worker-0/INFO]: AdvancedPeripherals says hello!
[00:18:46] [modloading-worker-0/INFO]: Forge mod loading, version 38.0.14, for MC 1.18 with MCP 20211130.085255
[00:18:46] [modloading-worker-0/INFO]: MinecraftForge v38.0.14 Initialized
[06Dec2021 00:18:47.073] [modloading-worker-0/ERROR] [net.minecraftforge.fml.javafmlmod.FMLModContainer/LOADING]: Failed to create mod instance. ModID: advancedperipherals, class de.srendi.advancedperipherals.AdvancedPeripherals
java.lang.reflect.InvocationTargetException: null
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:?]
	at java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[?:?]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[?:?]
	at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:81) ~[javafmllanguage-1.18-38.0.14.jar%2356!:?]
	at net.minecraftforge.fml.ModContainer.lambda$buildTransitionHandler$4(ModContainer.java:120) ~[fmlcore-1.18-38.0.14.jar%2355!:?]
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) [?:?]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) [?:?]
	at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) [?:?]
	at java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) [?:?]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) [?:?]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) [?:?]
Caused by: java.lang.NoSuchMethodError: 'java.lang.String net.minecraft.resources.ResourceLocation.getPath()'
	at de.srendi.advancedperipherals.common.setup.CCRegistration.<clinit>(CCRegistration.java:17) ~[advancedperipherals-1.18-0.7.7r.jar%2352!:0.7.7r]
	at de.srendi.advancedperipherals.common.setup.Registration.register(Registration.java:44) ~[advancedperipherals-1.18-0.7.7r.jar%2352!:0.7.7r]
	at de.srendi.advancedperipherals.AdvancedPeripherals.<init>(AdvancedPeripherals.java:53) ~[advancedperipherals-1.18-0.7.7r.jar%2352!:0.7.7r]
	... 14 more
[06Dec2021 00:18:47.758] [Render thread/FATAL] [net.minecraftforge.fml.ModLoader/LOADING]: Failed to complete lifecycle event CONSTRUCT, 1 errors found
```

### Steps to reproduce

1. Install Forge `38.0.14` (Minecraft 1.18).
2. Install Advanced Peripherals `0.7.7r`.
3. Start Minecraft.

After that, the game will crash

### Multiplayer?

False

### Version

0.7.7r

### Minecraft, Forge and maybe other related mods versions

Forge 38.0.14

### Crashlog/log

https://pastebin.com/xQdbmYxG

**Comment by Olfi01** [comment]
Correct me if I'm wrong, but the issue here seems to be that the current version of Advances Peripherals isn't compatible with CC: Tweaked 0.99 and upwards yet. Try downgrading CC: Tweaked to a version below that and get back to me with the results!

**Comment by SquidDev** [comment]
CC:T 1.99 is the only version for 1.18, so downgrading might be a little difficult :p. That AP is trying to use `ResourceLocation.getPath()` makes me wonder if the deobfuscated version of the mod was uploaded by accident.

**Comment by Olfi01** [comment]
Sorry about that, I misread the issue - somehow thought this was about the 1.16 issue I encountered myself a few days ago

**Comment by SirEndii** [comment]
I'm a bit confused about this issue. But I'm currently not at my computer, I will fix this tomorrow.

**Comment by BenjaminHoegh** [comment]
Same issue for me https://pastebin.com/n6LTNDG4

**Comment by SirEndii** [comment]
Okay, I now know what the issue is.
At the time we register the upgrades, the classes for the ID's are not available.

https://github.com/Seniorendi/AdvancedPeripherals/blob/1.18/src/main/java/de/srendi/advancedperipherals/common/setup/CCRegistration.java
A good question is, why does this work in the IDE dev environment.

**Comment by SirEndii** [comment]
Fixed, I will roll out the update today or tomorrow. Depends on how long I need to port the AE2 integration.

**Comment by SirEndii** [comment]
Published a new 1.18 version
